File: 백준/Silver/1764. 듣보잡/듣보잡.py
import sys
input = sys.stdin.readline

N,M = map(int,input().split())


lista = []
listb = []

for _ in range(N):
    lista.append(input().rstrip())
s1 = set(lista)

for _ in range(M):
    listb.append(input().rstrip())
s2 = set(listb)



dontseedonthear = list(s1.intersection(s2))

dontseedonthear.sort()


print(len(dontseedonthear))

for name in dontseedonthear:
    print(name)

File: 백준/Silver/1764. 듣보잡/듣보잡.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");
let [n, m] = input[0].split(" ").map(Number);
let set = new Set();

for (let i = 1; i <= n; i++) {
  set.add(input[i]);
}

let nameList = [];
for (let i = n + 1; i <= n + m; i++) {
  if (set.has(input[i])) {
    nameList.push(input[i]);
  }
}
nameList.sort();

console.log(nameList.length);
console.log(nameList.join("\n"));
